IT FIM (tier 2, Linux/CodeBuild): whodata tests not skipped — 83 failures after the CodeBuild migration

## Context

During the [Release 4.14.6 - RC 2 - Agent integration tests](https://github.com/wazuh/wazuh/issues/37236) run, the
`4_testintegration_fim-tier-2-lin` job failed:

- Run: https://github.com/wazuh/wazuh/actions/runs/28255178161
- Result in the RC table: 🔴 (see #37236 comment with per-test job links).

Result: **83 failed, 229 passed, 213 deselected, 1 xfailed**. **All 83 failures are
whodata-mode cases.**

This is a direct consequence of migrating the FIM Linux test workflows to AWS CodeBuild
(parent #37027, 4.x branch
`fix/37027-gha-4x-migrate-agent-linux-workflows-to-aws-codebuild-runners`; 5.x
equivalent PR #37051). The CodeBuild containers cannot provide the kernel audit subsystem
that FIM whodata depends on — and unlike tier 0-1, **the whodata filter was never applied
to the tier-2 job**, so all whodata cases run and fail.

## Root cause

FIM `whodata` mode relies on the Linux kernel netlink audit socket (`NETLINK_AUDIT`).
`syscheckd` calls `auditctl` to register rules; inside CodeBuild containers the required
capabilities (`CAP_AUDIT_CONTROL`, `CAP_AUDIT_WRITE`, `CAP_AUDIT_READ`) are absent
(blocked by the CodeBuild seccomp profile), so `auditctl` exits 255 and `syscheckd`
silently falls back from `whodata` to `realtime`. Tests asserting `mode == 'whodata'`
then fail. This is the same limitation documented for tier 0-1 in
`issue-it-fim-whodata-codebuild-no-audit.md`.

**The miss:** when whodata was filtered out of the tier-0-1 Linux job during the
migration, the **tier-2 Linux job was left running the whodata cases**. The tier-0-1
workflow filters them:

```yaml
# .github/workflows/4_testintegration_fim-tier-0-1-lin.yml
sudo python -m pytest --tier 0 --tier 1 test_fim/ \
  -k "not whodata and not Who-data" \
  --ignore=test_fim/test_files/test_follow_symbolic_link/test_audit_rules_with_symlink.py
```

…but the tier-2 workflow does not:

```yaml
# .github/workflows/4_testintegration_fim-tier-2-lin.yml
sudo python -m pytest --tier 2 test_fim/
```

## Failing tests (sample of the 83)

```
test_fim/test_files/test_ambiguous_confs/test_whodata_ambiguous_thread.py::test_whodata_ambiguous_thread[whodata_thread_enabled]                 - assert False == True
test_fim/test_files/test_ambiguous_confs/test_whodata_works_over_realtime.py::test_whodata_works_over_realtime[...]                                - AssertionError: assert 'realtime' == 'whodata'
test_fim/test_files/test_ignore/test_ignore_valid.py::test_ignore_subdirectory[not_ignore_sregex_whodata]                                          - AssertionError: FIM mode not equal
test_fim/test_files/test_ignore/test_ignore_valid.py::test_ignore_subdirectory[not_ignore_*_whodata*]   (many parametrizations)                   - AssertionError: FIM mode not equal
```

All share the same signature: the case expects `whodata` but the agent reports
`realtime` because the audit subsystem is unavailable.


## References

- Release test issue: #37236 (comment with per-test job links).
- Failing run: https://github.com/wazuh/wazuh/actions/runs/28255178161
- Parent epic: #37027 — migrate agent workflows to AWS CodeBuild.
- 5.x agent test migration: PR #37051.
- whodata root cause (tier 0-1): `issue-it-fim-whodata-codebuild-no-audit.md`.
- whodata coverage restoration: `issue-fim-whodata-skipped-codebuild.md`.
- DevOps / CodeBuild capability gaps: https://github.com/wazuh/wazuh-automation/issues/3497.

--- a/.github/workflows/4_testintegration_fim-tier-2-lin.yml
+++ b/.github/workflows/4_testintegration_fim-tier-2-lin.yml
@@ -88,4 +88,5 @@ jobs:
       - name: Run fim integration tests
         run: |
           cd tests/integration
-          sudo python -m pytest --tier 2 test_fim/
+          sudo python -m pytest --tier 2 test_fim/ \
+            -k "not whodata and not Who-data"

